src/aliased_buffer.h

```diff
@@ -0,0 +1,191 @@
+ * going through JS, and the data is then available to user's via the exposed
+ * JS object.
+ *
+ * While this technique is computationaly efficient, it is effectively a
+ * write to JS program state w/out going through the standard
+ * (monitored) API. Thus any VM capabilities to detect the modification are
+ * circumvented.
+ *
+ * The encapsulation herein provides a placeholder where such writes can be
+ * observed. Any notification APIs will be left as a future exercise.
+ */
+template <class NativeT, class V8T>
+class AliasedBuffer {
+ public:
+  AliasedBuffer(v8::Isolate* isolate, const size_t count) :
+      isolate_(isolate),
+      count_(count),
+      byte_offset_(0),
+      free_buffer_(true) {
+    const v8::HandleScope handle_scope(isolate_);
+
+    const size_t sizeInBytes = sizeof(NativeT) * count;
+
+    // allocate native buffer
+    buffer_ = Calloc<NativeT>(count);
+
+    // allocate v8 ArrayBuffer
+    v8::Local<v8::ArrayBuffer> ab = v8::ArrayBuffer::New(
+        isolate_, buffer_, sizeInBytes);
+
+    // allocate v8 TypedArray
+    v8::Local<V8T> js_array = V8T::New(ab, byte_offset_, count);
+    js_array_ = v8::Global<V8T>(isolate, js_array);
+  }
+
+  /**
+   * Create an AliasedBuffer over a sub-region of another aliased buffer.
+   * The two will share a v8::ArrayBuffer instance &
+   * a native buffer, but will each read/write to different sections of the
+   * native buffer.
+   *
+   *  Note that byte_offset must by aligned by sizeof(NativeT).
+   */
+  AliasedBuffer(
+      v8::Isolate* isolate,
+      const size_t byte_offset,
+      const size_t count,
+      const AliasedBuffer<uint8_t,
+      v8::Uint8Array>& backing_buffer) :
+      isolate_(isolate),
+      count_(count),
+      byte_offset_(byte_offset),
+      free_buffer_(false) {
+    const v8::HandleScope handle_scope(isolate_);
+    // validate that the byte_offset is aligned with sizeof(NativeT)
+    CHECK_EQ(byte_offset & (sizeof(NativeT) - 1), 0);
+    buffer_ = reinterpret_cast<NativeT*>(
+        const_cast<uint8_t*>(backing_buffer.GetNativeBuffer() + byte_offset));
+
+    v8::Local<v8::ArrayBuffer> ab = backing_buffer.GetArrayBuffer();
+    v8::Local<V8T> js_array = V8T::New(ab, byte_offset, count);
+    js_array_ = v8::Global<V8T>(isolate, js_array);
+  }
+
+  AliasedBuffer(const AliasedBuffer& that) :
+      isolate_(that.isolate_),
+      count_(that.count_),
+      byte_offset_(that.byte_offset_),
+      buffer_(that.buffer_),
+      free_buffer_(false) {
+    js_array_ = v8::Global<V8T>(that.isolate_, that.GetJSArray());
+  }
+
+  ~AliasedBuffer() {
+    if (free_buffer_ && buffer_ != NULL) {
+      free(buffer_);
+    }
+    js_array_.Reset();
+  }
+
+  /**
+   * Helper class that is returned from operator[] to support assignment into
+   * a specified location.
+   */
+  class Reference {
+   public:
+    Reference(AliasedBuffer<NativeT, V8T>* aliased_buffer, size_t index) :
+      aliased_buffer_(aliased_buffer),
```

Comment:
please at least make the indentation style within the same file consistent. the majority of places in core (which I thought was part of the linter because we've been fairly consistent) would look like:
```cc
Reference(AliasedBuffer<NativeT, V8T>* aliased_buffer, size_t index)
  : aliased_buffer_(aliased_buffer),
    index_(index) {
```